src/NCronJob/Configuration/NCronJobOptionBuilder.cs

```diff
@@ -154,51 +162,183 @@ internal static string GenerateJobName(Delegate jobDelegate)
         var jobHash = jobNameBuilder.ToString().GenerateConsistentShortHash();
         return $"AnonymousJob_{jobHash}";
     }
+
+    internal void RegisterJobs()
+    {
+        foreach (var job in jobs)
+        {
+            Services.AddSingleton(job);
```

Comment:
Shall we use `TryAddSingleton`?

Reply:
This registration is a replacement for this:
```csharp
foreach (var option in jobOptions)
{
    var cron = option.CronExpression is not null
        ? GetCronExpression(option)
        : null;
    var entry = new JobDefinition(typeof(T), option.Parameter, cron, option.TimeZoneInfo);
--->Services.AddSingleton(entry);
}
```

So I don't think we want to do that otherwise you wont be able to register multiple schedules per Job type.